### DISCUSSION TITLE
Feature request: Use system locale for number formats or make thousands and decimal separators configurable

### DISCUSSION THREAD

**Discussion by Eucyon** [andmarti1424/sc-im#641] [Ideas]
In some languages thousands and decimal separators are different from the English ones, e.g.:

English: 123,456.789 ~ German: 123.456,789.

Could you please enable this in sc-im, maybe via using locales or configuration?

Thank you!

**Comment by andmarti1424** [comment]
Do you know if its supported on sc?
Move to discussion so others can give their opinion on this matter.

**Comment by Eucyon** [comment]
```sc-im --version```

in the terminal gives

```SC-IM - version 0.8.2
-DNCURSES
-DMAXROWS 65536
-DUNDO
-DXLSX
-DDEFAULT_COPY_TO_CLIPBOARD_CMD="xclip -i -selection clipboard <"
-DDEFAULT_PASTE_FROM_CLIPBOARD_CMD="xclip -o -selection clipboard"
-DDEFAULT_OPEN_FILE_UNDER_CURSOR_CMD="scopen"
-DUSELOCALE
-DMOUSE
-DUSECOLORS
-D_XOPEN_SOURCE_EXTENDED
-D_GNU_SOURCE
-DSNAME="sc-im"
-DHELP_PATH="/usr/share/sc-im"
-DLIBDIR="/usr/share/doc/sc-im"
-DDFLT_PAGER="less"
-DDFLT_EDITOR="vim"
-DCONFIG_DIR=".config/sc-im"
-DCONFIG_FILE="scimrc"
-DHISTORY_DIR=".cache"
-DHISTORY_FILE="sc-iminfo"
-DINS_HISTORY_FILE="sc-iminfo"
-DHAVE_PTHREAD
-DAUTOBACKUP
```

so something with locales is compiled in sc-im.

```
man sc
```

on Arch Linux shows from up line 491

```
              locale/!locale
                     If  locale  support  is compiled into sc, this option will cause certain lo‐
                     cale-dependent behaviors, such as the display of numbers and the  determina‐
                     tion of word boundaries for some operations in edit mode.  Note that if this
                     option is set and the environment variable LC_ALL is unrecognized, unset, or
                     set to either ``POSIX'' or ``C'', commas in format commands will be ignored.
```

Although my locales are set to German, numbers in sc-im have the format 123,456.789 instead of 123.456,789.

```
set locale
```
in sc-im's configuration file gives a "syntax error" while start up.

**Reply by andmarti1424** [reply]
And if you try entering 123.456,789 on sc? Does it work?

**Reply by Eucyon** [reply]
No, sc-im says "syntax error" and shows "123.456".

**Comment by Eucyon** [comment]
Sorry, tried it in sc-im. But sc behaves in the same way.

**Comment by 1dancook** [comment]
It will be good to have locale dependent number formats.

If someone working in Germany wants to share a sheet with an American team, having the locale respected for how numbers are displayed on sounds good. The internal representation can be the same.

**Comment by geckopfote** [comment]
I would like to bring up this topic again. I also work with German localization. When entering many numbers in succession, it is inconvenient not to be able to use the familiar comma  (",") as a decimal separator. You have to switch back and forth between the numeric and character keypad on the keyboard.

**Comment by andmarti1424** [comment]
Tried, but I am afraid it is not as easy as setting LC_NUMERIC in locale. lex.c has to be changed in a lot of places and probably in gram.y too.

**Comment by andmarti1424** [comment]
Confirmed. You have to change lex.c, install proper locales in system. Make sure you got them and then it might work, like this:
<img width="1366" height="248" alt="image" src="https://github.com/user-attachments/assets/db568b61-77d1-4042-a0ca-ae46fe6814a3" />
the thing is that you depend on having the locale installed, otherwise you might ( most certainly will) get wrong results..

**Comment by andmarti1424** [comment]
In case you want to test this... 
[641.patch](https://github.com/user-attachments/files/22348958/641.patch)
